src/Evolution/DiscontinuousGalerkin/Limiters/Weno.hpp

```diff
@@ -339,6 +340,21 @@ bool Weno<VolumeDim, tmpl::list<Tags...>>::operator()(
       modified_neighbor_solutions.insert(
           std::make_pair(neighbor, interpolant.interpolate(data.volume_data)));
     }
+
+    // Then the correction is added one tensor component at a time:
+    const auto wrap_mean_correction = [&mesh, &modified_neighbor_solutions ](
+        auto tag, const auto& tensor) noexcept {
+      for (size_t i = 0; i < tensor->size(); ++i) {
+        const double mean_local = mean_value((*tensor)[i], mesh);
+        for (auto& kv : modified_neighbor_solutions) {
+          DataVector& component_to_correct = get<decltype(tag)>(kv.second)[i];
```

Comment:
I would've found this code slightly easier to parse had the naming been `nhbr_component_to_correct` and `mean_nhbr`. In the helpers file below you name them `local_mean` and `neighbor_mean`. I'm not picky about the exact wording or consistency across files, but would briefer something indicating `neighbor` in the name :)